textile/fabric_printing/doctype/coating_order/coating_order.py

```diff
@@ -13,18 +13,26 @@
 
 
 force_fields = ["customer_name", "fabric_item_name", "fabric_material",
-				"fabric_type", "fabric_width", "fabric_gsm", "fabric_per_pickup"]
+	"fabric_type", "fabric_width", "fabric_gsm", "fabric_per_pickup"]
 
 
 class CoatingOrder(TextileOrder):
+	@property
+	def fabric_stock_qty(self):
+		from erpnext.stock.get_item_details import get_bin_details
+		return get_bin_details(self.fabric_item, self.fabric_warehouse).get("actual_qty") or 0
+
 	def onload(self):
 		self.set_fabric_stock_qty()
+		if self.docstatus == 0:
+			self.set_missing_values()
+			self.calculate_totals()
 
 	def validate(self):
 		self.set_missing_values()
 		self.validate_dates()
 		self.validate_customer()
-		self.validate_fabric_item()
+		TextileOrder.validate_fabric_item("Ready Fabric")
```

Comment:
we still use `self.` notation even if it's a staticmethod, for the simpliciy of changing staticmethod to object method in the future and consistency as well